integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/End2EndTestITStep.java

```diff
@@ -0,0 +1,272 @@
+import static com.redhat.service.bridge.integration.tests.common.BridgeCommon.getProcessor;
+import static io.restassured.RestAssured.given;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class End2EndTestITStep extends AbstractBridge {
+
+    private static final String TOPIC_NAME = "myKafkaTopic";
+    private static final Set<BaseFilter> FILTERS = Collections.singleton(new StringEquals("source", "StorageService"));
+    private static final String TRANSFORMATION_TEMPLATE = "{\"v1\": \"{data.k1}\"}";
+    private static final String PROCESSOR_NAME = "myProcessor";
+    private static String BRIDGE_NAME = "notificationBridge";
+
+    private static String bridgeId;
+    private static String processorId;
+    private static String endPoint;
+    private static String ingressMetrics;
+    private static ProcessorRequest processorRequest;
+
+    @Given("Manager url is not accessible")
+    public void authenticationIsEnabled() {
+        given()
+                .filter(new ResponseLoggingFilter())
+                .contentType(ContentType.JSON)
+                .when()
+                .get(managerUrl + APIConstants.USER_API_BASE_PATH)
+                .then()
+                .statusCode(401);
+    }
+
+    @Given("Bridge list is empty")
+    public void getEmptyBridges() {
+
+        BridgeListResponse response = getBridgeList();
+        assertThat(response.getKind()).isEqualTo("BridgeList");
+        assertThat(response.getSize()).isZero();
+        assertThat(response.getPage()).isZero();
+        assertThat(response.getTotal()).isZero();
+    }
+
+    @When("Create a Bridge")
+    public void createBridge() {
+        BridgeResponse response = addBridge(BRIDGE_NAME);
+        assertThat(response.getName()).isEqualTo(BRIDGE_NAME);
+        assertThat(response.getStatus()).isEqualTo(BridgeStatus.REQUESTED);
+        assertThat(response.getEndpoint()).isNull();
+        assertThat(response.getPublishedAt()).isNull();
+        assertThat(response.getHref()).isNotNull();
+        assertThat(response.getSubmittedAt()).isNotNull();
+
+        bridgeId = response.getId();
+    }
+
+    @Then("^Bridge is exists in status \"([^\"]*)\" within (\\d+) (?:minute|minutes)$")
+    public void bridgeDeployedWithinMinutes(String status, int timeoutMinutes) {
+        Awaitility.await()
+                .atMost(Duration.ofMinutes(timeoutMinutes))
+                .pollInterval(Duration.ofSeconds(5))
+                .untilAsserted(
+                        () -> getBridgeDetails(bridgeId)
+                                .then()
+                                .body("status", Matchers.equalTo(status))
+                                .body("endpoint", Matchers.containsString(bridgeId)));
+
+        // store bridge endpoint details
+        endPoint = getBridgeDetails(bridgeId).then().extract().response().as(BridgeResponse.class).getEndpoint();
+    }
+
+    @When("Add processor to the bridge")
```

Comment:
```suggestion
    @When("^add Processor to the Bridge with access token:$")
```
with request body in feature file like:
```
{
  "name": "myProcessor",
  "action": {
    "name": "myKafkaAction",
    "parameters": {"topic":  "demoTopic"},
    "type": "KafkaTopicAction"
  },
  "filters": [
    {
      "key": "source",
      "type": "StringEquals",
      "value": "StorageService"
    }
  ]
}
```